src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -640,7 +640,7 @@ public FeatureCollectionResponse run(GetFeatureRequest request)
                 if (!isNumberMatchedSkipped) {
                         if (calculateSize
                                 && (queryMaxFeatures == Integer.MAX_VALUE || size < queryMaxFeatures)
-                                && offset <= 0) {
+                                && offset <= 0 && !isV_20) {
```

Comment:
Can you elaborate what in WFS 2.0 makes things different than in WFS 1.1? I cannot remember off the top of my head, maybe it's best to also clarify it in the comment (here and below)